specifications/terminology.md

```diff
@@ -0,0 +1,3 @@
+# Terminology
```

Comment:
We might have to move some definitions from `dataspace.ecosystem.md` to here in the future but for now, I don't mind adding it empty.

Reply:
Yes, agreed.